### PR TITLE
update logic to read addon templates from metadata

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/kontainer-driver-metadata/pull/19

### CODE REVIEW

File: cluster/network.go [hunk 1]
```diff
@@ -245,14 +245,12 @@ func (c *Cluster) doWeaveDeploy(ctx context.Context, data map[string]interface{}
...
 	switch c.Network.Plugin {
-	case FlannelNetworkPlugin:
-		return templates.CompileTemplateFromMap(templates.GetVersionedTemplates(FlannelNetworkPlugin, data, c.Version), pluginConfig)
-	case CalicoNetworkPlugin:
-		return templates.CompileTemplateFromMap(templates.GetVersionedTemplates(CalicoNetworkPlugin, data, c.Version), pluginConfig)
-	case CanalNetworkPlugin:
-		return templates.CompileTemplateFromMap(templates.GetVersionedTemplates(CanalNetworkPlugin, data, c.Version), pluginConfig)
-	case WeaveNetworkPlugin:
-		return templates.CompileTemplateFromMap(templates.GetVersionedTemplates(WeaveNetworkPlugin, data, c.Version), pluginConfig)
+	case FlannelNetworkPlugin, CalicoNetworkPlugin, WeaveNetworkPlugin:
```
Comment (superseb) [inline cluster/network.go]: Is this missing `CanalNetworkPlugin` on purpose?

Comment (kinarashah) [reply cluster/network.go]: No, removed it by mistake, updated

**Comment by prachidamle** [comment]
LGTM

**Comment by superseb** [comment]
LGTM

**Pull request merged**